index.js

```diff
@@ -37,7 +35,7 @@ async function walk(output, prefix, lexer, opts, dirname='', level=0) {
     if (rgx && !rgx.test(file)) continue;
     !filesOnly && isMatch && output.push(join(prefix, relpath));
 
-    await walk(output, prefix, lexer, opts, relpath, giveup(rgx) ? null : level + 1);
+    await walk(output, prefix, lexer, opts, relpath, !rgx || `${rgx}` == lexer.globstar ? null : level + 1);
```

Comment:
Let's make this:

```js
rgx && rgx.toString() !== lexer.globster && ++level
```

Passing along `null` doesn't do anything wrt the default argument value -- same with `false`. And using `.toString()` variant is a little more obvious / harder to miss when reading later.